### DISCUSSION TITLE
[Colorschemes] How to remove unwanted item from the list of colorschemes.

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#3695] [Q&A] [answered]
I want to keep only a few I want and not show them all.

![image](https://github.com/LazyVim/LazyVim/assets/29248764/84d272e7-af7e-40ba-b60c-4773b97c5ca8)

**Comment by dpetka2001** [comment]
I used to have a function as can be seen [here in my dots](https://github.com/dpetka2001/dotfiles/blob/ca6bff6219348b95866502584ee75b206a5b4274/dot_config/nvim/lua/plugins/colorscheme.lua#L1-L24). Recently, I tried this in a testing configuration 
```lua
local builtins = {}

for name, _ in vim.fs.dir("$VIMRUNTIME/colors/") do
  if name:match("*.txt") then
    break
  else
    name = name:match(".*%.lua$") or name:match(".*%.vim$")
    table.insert(builtins, name)
  end
end

vim.opt.wildignore:append(builtins)
```
in my `/lua/config/options.lua` and I think I'll also update my personal configuration to this. Both of them do the same thing, ignoring the default Neovim colorschemes.

**Reply by iamxiaojianzheng** [reply, marked as answer]
Based on your code, I implemented this idea that you are really cool.

```lua
local get_colorsheme = function()
  local builtins = {}

  for name, _ in vim.fs.dir("$VIMRUNTIME/colors/") do
    if name:match("*.txt") then
      break
    else
      name = name:match("(.*).lua$") or name:match("(.*).vim$")
      table.insert(builtins, name)
    end
  end
  local target = vim.fn.getcompletion

  ---@diagnostic disable-next-line: duplicate-set-field
  vim.fn.getcompletion = function()
    return vim.tbl_filter(function(color)
      return not vim.tbl_contains(builtins, color)
      ---@diagnostic disable-next-line: param-type-mismatch
    end, target("", "color"))
  end

  LazyVim.pick("colorschemes")()
  vim.fn.getcompletion = target
end

return {
  {
    "ellisonleao/gruvbox.nvim",
    keys = {
      { "<leader>uC", get_colorsheme, desc = "Colorscheme with preview" },
    },
  },
  { "sainnhe/gruvbox-material" },
  {
    "rebelot/kanagawa.nvim",
    keys = {
      { "<leader>uC", get_colorsheme, desc = "Colorscheme with preview" },
    },
  },
  {
    "folke/tokyonight.nvim",
    lazy = false,
    priority = 1000,
    opts = {},
    keys = {
      { "<leader>uC", get_colorsheme, desc = "Colorscheme with preview" },
    },
  },
  -- disabled default keymap
  {
    "ibhagwan/fzf-lua",
    event = "VeryLazy",
    keys = {
      { "<leader>uC", false },
    },
  },
  -- Configure LazyVim to load gruvbox
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "gruvbox",
    },
  },
}
```

![image](https://github.com/LazyVim/LazyVim/assets/29248764/ad033b07-9987-4ad4-af3a-16f9e2475146)

**Comment by iamxiaojianzheng** [comment]
@dpetka2001 

I would like to ask an additional question about lazyvim's reload function. 

If I modify and save plugin/xxx.lua, I will be prompted to reload this configuration in the upper right corner. 

But the actual effect is not using the latest configuration, so what does this reload really do?

**Reply by dpetka2001** [reply]
The reload notification that you're seeing is the detection from `lazy.nvim` when you change some configuration file. So, if you remove a plugin spec, then in Lazy UI you'll see the change that the plugin is ready to be removed. 

It doesn't reload your Neovim configuration. There's an option in `lazy.nvim` about reloading a plugin spec but [Folke himself](https://github.com/folke/lazy.nvim/issues/1480#issuecomment-2133299067) said it doesn't work 100%. So, your best bet is to still restart Neovim for the latest changes in your configuration to take effect.

If you don't like that notification you can disable it. In your `/lua/config/lazy.lua`
```lua
change_detection = {
    -- automatically check for config file changes and reload the ui
    enabled = true,
    notify = true, -- get a notification when changes are found
  },
```
Change `notify = false` if you don't want to see these notifications.

**Reply by iamxiaojianzheng** [reply]
@dpetka2001 

I have another question to ask you.

I found that my stylua.toml didn't seem to work, such as the 'column_width = 120' configuration item, even if the lua code line is less than 120 characters, will be formatted.

**Reply by dpetka2001** [reply]
I'm not familiar with this. Sorry.

**Reply by iamxiaojianzheng** [reply]
I know the reason, LazyVim source code is through this to control the prohibition of formatting

`-- stylua: ignore`

**Comment by abeldekat** [comment]
Recently, telescope added the ignore_builtins option:

https://github.com/nvim-telescope/telescope.nvim/pull/3155

It's now as simple as `Telescope colorscheme ignore_builtins=true`

**Reply by dpetka2001** [reply]
I've switched to fzf-lua instead of Telescope. I just have this
```lua
local builtins = {}

for name, _ in vim.fs.dir("$VIMRUNTIME/colors/") do
  if name:match("*.txt") then
    break
  else
    name = name:match(".*%.lua$") or name:match(".*%.vim$")
    table.insert(builtins, name)
  end
end

vim.opt.wildignore:append(builtins)
```
in my `options.lua` instead of a function that gets passed into telescope keymap. You also don't have to override `vim.fn.completion` temporarily :stuck_out_tongue:

**Reply by dpetka2001** [reply]
I don't get completions about searches from those files with this, but it's something I'm willing to live with since I'm not interested in them.

**Reply by abeldekat** [reply]
That's a nice solution!
As an alternative, I think you could also add the ignored `builtins` to the `FzfLua colorscheme` command, using opt [`ignore_patterns`](https://github.com/ibhagwan/fzf-lua/blob/c56717d6833166adadd11cad05d979b1fb370c57/lua/fzf-lua/providers/colorschemes.lua#L30).
That way, you don't need to "wildignore".

**Reply by dpetka2001** [reply]
Yes, I know. I just prefer it this way since it's independent of which filepicker you use (of course with `LazyVim.pick` you can easily incorporate it regardless of which filepicker) and you don't have to pass anything to whatever function. Like I said I just don't care about these files at all, so I'm personally willing to not get any results about them anywhere.
